labs/lab-05/tasks/vec-count-if/solution/count_pos_neg.asm

```diff
@@ -1,30 +1,43 @@
 ; SPDX-License-Identifier: BSD-3-Clause
 
-%include "../utils/printf32.asm"
+%include "printf64.asm"
 
 %define ARRAY_SIZE    10
 
 section .data
-    dword_array dd 1392, -12544, -7992, -6992, 7202, 27187, 28789, -17897, 12988, 17992
+    qword_array dq 1392, -12544, -7992, -6992, 7202, 27187, 28789, -17897, 12988, 17992
+    fmt_str db "Num pos is %lu, num neg is %lu", 10, 0
 
 section .text
 extern printf
 global main
 main:
-    mov ecx, ARRAY_SIZE     ; Use ecx as loop counter.
-    xor ebx, ebx            ; Store positive number in ebx.
-    xor edx, edx            ; Store negative number in edx.
+    push rbp
+    mov rbp, rsp
+
+    mov rcx, ARRAY_SIZE     ; Use rcx as loop counter.
+    xor rbx, rbx            ; Store positive number in rbx.
+    xor rdx, rdx            ; Store negative number in rdx.
 next_element:
-    mov eax, dword [dword_array + ecx*4 - 4]
-    cmp eax, 0
+    mov rax, qword [qword_array + rcx*8 - 8]
+    cmp rax, 0
     jl add_to_neg
-    inc ebx
+    inc rbx
     jmp test_end
 add_to_neg:
-    inc edx
+    inc rdx
 test_end:
-    loop next_element ; Decrement ecx, if not zero, go to next element.
+    loop next_element ; Decrement rcx, if not zero, go to next element.
+
+    ; Save rdx as it will be used by printf
+    push rdx
 
-    PRINTF32 `Num pos is %u, num neg is %u\n\x0`, ebx, edx
+    ; Call printf with proper x86_64 calling convention
+    mov rdi, fmt_str        ; First arg: format string
+    mov rsi, rbx            ; Second arg: positive count
+    pop rdx                 ; Third arg: negative count (restore from stack)
+    xor rax, rax            ; No vector registers used
+    call printf
```

Comment:
This is overkill and students aren't supposed to know how to call functions yet. `PRINTF64` works fine.